### DISCUSSION TITLE
Disable cell action based on value

### DISCUSSION THREAD

**Discussion by gotzsys** [filamentphp/filament#6148] [Help]
Hello guys
How can I disable the cell action if the status is "used"?

BadgeColumn::make('status')
                    ->label('Status')
                    ->enum(VoucherResource::FILTER_STATUS)
                    ->colors([
                        'primary' => 'not_used',
                        'success' => 'used',
                        'danger' => 'expired',
                    ])
                    ->toggleable()
                    ->action(
                        Action::make('updateStatus')
                            ->label('Atualizar status')
                            ->mountUsing(fn (Forms\ComponentContainer $form, Voucher $record) => $form->fill([
                                'status' => $record->status,
                            ]))
                            ->action(function (Voucher $record, array $data): void {
                                $record->status = $data['status'];
                                $record->used_at = ($data['status'] == 'used') ? now() : null;
                                $record->save();
                            })
                            ->form([
                                Select::make('status')
                                    ->required()
                                    ->options(VoucherResource::FILTER_STATUS)
                                    ->searchable(),
                            ])
                    ),

**Comment by danharrin** [comment]
Probably `->action(fn ($record) => // check record and conditionally return action)`

**Comment by gotzsys** [comment]
Hi Dan, I tryed :(
But the modal action not open even when status == 'pending'

BadgeColumn::make('status')
                    ->label('Status')
                    ->enum(self::FILTER_STATUS)
                    ->colors([
                        'primary' => 'pending',
                        'success' => 'confirmed',
                        'danger' => 'canceled',
                    ])
                    ->toggleable()
                    ->action(
                        function ($record) {
                            if ($record->status === 'pending') {
                                return Action::make('updateStatus')
                                    ->label('Atualizar status')
                                    ->mountUsing(fn (Forms\ComponentContainer $form, Loyalty $record) => $form->fill([
                                        'status' => $record->status,
                                    ]))
                                    ->action(function (Loyalty $record, array $data): void {
                                        $record->status = $data['status'];
                                        ($data['status'] === 'confirmed') ? $record->confirmed_at = now() : null;
                                        $record->save();
                                    })
                                    ->form([
                                        Select::make('status')
                                            ->required()
                                            ->options(self::FILTER_STATUS)
                                            ->searchable(),
                                    ]);
                            }
                        }
                    ),

I solved it by disabling the action:
->disabled(fn (Loyalty $record) => $record->status !== 'pending')

It's not exactly what I wanted, because the badge is still clickable, but it works...

**Comment by judgej** [comment]
You could use `disableClick()` as well as `disabled()`. That removes the click activity and hand mouse pointer. I think you still need to explicitely change the colour of the icon if you want it to be visibly disabled.
